### PR TITLE
Light node

### PR DESCRIPTION
**Pull request by vegkams**
Test before merge

### CODE REVIEW

**Comment by vegkams** [comment]
[Lys-node](https://trello.com/c/DK7ongbA/191-lys-node)

File: vortex_msgs/msg/LightInput.msg [hunk 1]
```diff
@@ -1 +1,4 @@
-int32 light_intensity
+# Message for toggling lights on ROV
+# List of strings contain the name of the light(s)
+string[] toggle_light
+uint8[] intensity
```
Comment (mortenfyhn) [inline vortex_msgs/msg/LightInput.msg]: Update documentation here. Mention that the indices of the lists correspond (if they do). Specify the range for intensity values. Is it 0-255?

**Review by mortenfyhn — CHANGES_REQUESTED** [review]
Looks good, overall. Let's test it on hw before merging.

**Review by mortenfyhn — APPROVED** [review]

File: light/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# light
+Node for controlling the ROV lights
```
Comment (sindrehan) [inline light/README.md]: A more extensive Readme would be nice.

Comment (sindrehan) [reply light/README.md]: E.g. add a line or two about the dependencies.

File: light/scripts/light_node.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+import rospy
+import Adafruit_BBIO.GPIO as GPIO
+from vortex_msgs.msg import LightInput, Pwm
+
+FULL_CYCLE = 4096
```
Comment (sindrehan) [inline light/scripts/light_node.py]: Should be 4095.

Comment (sindrehan) [reply light/scripts/light_node.py]: Consider loading parameter from manipulator branch.

File: light/scripts/light_node.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    def callback(self, msg):
+        self.light_control(msg)
+
+    def light_control(self, msg):
+        for index, light in enumerate(msg.toggle_light):
+            if light in PWM_PIN_MAP:
+                pwm_msg = Pwm()
+                Pwm.pins[0] = PWM_PIN_MAP[light]
+                Pwm.on[0] = 0
+                Pwm.off = (FULL_CYCLE * msg.intensity[index]) // 100
```
Comment (sindrehan) [inline light/scripts/light_node.py]: - Pwm.off is a list and need an index.

**Review by sindrehan — APPROVED** [review]

**Pull request merged**